keps/sig-api-machinery/2144-clientgo-apply/README.md

```diff
@@ -358,6 +284,66 @@ adding functions to interfaces is a change we have made in the past, and develop
 that have alternate implementations of the interface will usually get a compiler
 error in this case, which is relatively trivial to resolve
 
+### read/modify/write loop support
+
+While it is
+[recommended](https://kubernetes.io/docs/reference/using-api/server-side-apply/)
+that apply clients use "fully specified intent" rather than
+read/modify/write loops, there are many existing controllers that use
+a get/modify/update loop today, and are not easy to convert to the
+"fully specified intend" approach.
+
+For such controllers, providing a convenient way to do a
+"get-previously-applied/modify/apply" loop is pragmatic. It allows
+these controller to benefit from apply by sending a minimal apply
+patch (as opposed to sending the entire object for update) and to
+reduce the odds of conflict with other controllers.
+
+To support this, the builders will include "BuildApply" utility functions
+function. For example:
+
+```
+fieldManger := "my-field-manager"
+// 1. read
+deployment, err := client.AppsV1().Deployments(ns).Get("deployment-name", metav1.GetOptions{})
+if err != nil {
+  // handle err
+}
+applyConfig, err := appsv1apply.BuildDeploymentApply(deployment, fieldManager)
+if err != nil {
+  // handle err
+}
+
+// 2. modify
+applyConfig.GetSpec().Replicas(10)
+
+// 3. apply
+client.AppsV1().Deployments(ns).Apply(ctx, applyConfig, metav1.ApplyOptions{FieldManager: fieldManager, Force: true})
```

Comment:
I remember we originally created this apply function to incorporate the field manager as a separate parameter. It looks like we're reverting back to using the ApplyOptions, just want to confirm that we'll still use a new Apply function instead of a Patch with applypatch.

Reply:
I think we shouldn't have two ways of passing the field manager because then people could tell us different things and then what do we do?

Reply:
Yeah I agree. Was just pointing out that this function signature is *almost* identical to the Patch function (minus the need to specify a patch type) so it may be just for stylistic purposes to explicitly have an Apply function. I don't have any strong preferences.

Reply:
For client-go, the difference between Path and Apply is quite significant because patch takes a `data []byte` argument for the object where-as Apply takes a typed Apply Configuration.